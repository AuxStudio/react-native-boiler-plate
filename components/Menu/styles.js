import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {},
  itemContainer: {},
  itemText: {},
});

export default styles;
