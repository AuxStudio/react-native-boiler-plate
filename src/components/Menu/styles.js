import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {},
  itemContainer: {},
  itemText: {},
});

export default styles;
