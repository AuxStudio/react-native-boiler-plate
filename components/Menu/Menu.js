import React from 'react';
import PropTypes from 'prop-types';
import { ViewPropTypes, Text } from 'react-native';
import Menu, { MenuItem } from 'react-native-material-menu';
import { Touchable } from 'react-native-simple-components';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

export default class MenuComponent extends React.Component {
  static propTypes = {
    options: PropTypes.arrayOf(
      PropTypes.shape({
        text: PropTypes.string.isRequired,
        disabled: PropTypes.bool,
      }),
    ).isRequired,
    handlePress: PropTypes.func.isRequired,
    itemTextStyle: Text.propTypes.style,
    itemContainerStyle: ViewPropTypes.style,
    containerStyle: ViewPropTypes.style,
  };

  static defaultProps = {};

  state = {};

  setMenuRef = (ref) => {
    this.menu = ref;
  };

  menu = null;

  showMenu = () => {
    this.menu.show();
  };

  selectOption = (option) => {
    if (this.props.handlePress) {
      this.props.handlePress(option);
    }
    this.hideMenu();
  };

  hideMenu = () => {
    this.menu.hide();
  };

  render() {
    const menuButtonComponent = (
      <Touchable onPress={this.showMenu} style={styles.menuButton}>
        <Icon name="menu" style={styles.menuButtonIcon} />
      </Touchable>
    );

    return (
      <Menu
        ref={this.setMenuRef}
        button={menuButtonComponent}
        style={[styles.container, this.props.containerStyle]}
      >
        {this.props.options &&
          this.props.options.map((option) => {
            return (
              <MenuItem
                key={option.text}
                onPress={() => this.selectOption(option.text)}
                disabled={option.disabled}
                textStyle={[styles.itemText, this.props.itemTextStyle]}
                style={[styles.itemContaier, this.props.itemContainerStyle]}
              >
                {option.text}
              </MenuItem>
            );
          })}
      </Menu>
    );
  }
}
