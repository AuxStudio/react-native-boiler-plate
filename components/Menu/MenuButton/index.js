import React from 'react';
import PropTypes from 'prop-types';
import { Text, ViewPropTypes } from 'react-native';
import { Touchable } from 'react-native-simple-components';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

const propTypes = {
  iconName: PropTypes.string,
  handlePress: PropTypes.func,
  iconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
};

const defaultProps = {
  iconName: 'menu',
};

const MenuButton = ({ iconName, handlePress, iconStyle, style }) => {
  return (
    <Touchable onPress={handlePress} style={[styles.menuButton, style]}>
      <Icon name={iconName} style={[styles.menuButtonIcon, iconStyle]} />
    </Touchable>
  );
};

MenuButton.propTypes = propTypes;
MenuButton.defaultProps = defaultProps;

export default MenuButton;
