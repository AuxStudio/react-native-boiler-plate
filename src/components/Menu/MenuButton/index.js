import React from 'react';
import PropTypes from 'prop-types';
import { Text, ViewPropTypes } from 'react-native';
import { Touchable } from 'react-native-simple-components';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

const propTypes = {
  iconName: PropTypes.string,
  handlePress: PropTypes.func,
  iconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
  testID: PropTypes.string,
};

const defaultProps = {
  iconName: 'menu',
};

const MenuButton = ({ iconName, handlePress, iconStyle, style, testID }) => {
  return (
    <Touchable onPress={handlePress} style={[styles.menuButton, style]} testID={testID}>
      <Icon name={iconName} style={[styles.menuButtonIcon, iconStyle]} />
    </Touchable>
  );
};

MenuButton.propTypes = propTypes;
MenuButton.defaultProps = defaultProps;

export default MenuButton;
