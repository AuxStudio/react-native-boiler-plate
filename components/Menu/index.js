import React from 'react';
import PropTypes from 'prop-types';
import { ViewPropTypes, Text } from 'react-native';
import Menu, { MenuItem } from 'react-native-material-menu';

import styles from './styles';

import MenuButton from './MenuButton';

export default class MenuComponent extends React.Component {
  constructor(props) {
    super(props);

    this.setMenuRef = this.setMenuRef.bind(this);
    this.showMenu = this.showMenu.bind(this);
    this.selectOption = this.selectOption.bind(this);
    this.hideMenu = this.hideMenu.bind(this);

    this.menu = null;

    this.state = {};
  }

  static propTypes = {
    options: PropTypes.arrayOf(
      PropTypes.shape({
        text: PropTypes.string.isRequired,
        disabled: PropTypes.bool,
      }),
    ).isRequired,
    handlePress: PropTypes.func.isRequired,
    iconName: PropTypes.string,
    iconStyle: Text.propTypes.style,
    iconContainerStyle: ViewPropTypes.style,
    itemTextStyle: Text.propTypes.style,
    itemContainerStyle: ViewPropTypes.style,
    containerStyle: ViewPropTypes.style,

    buttonTestID: PropTypes.string,
  };

  static defaultProps = {};

  setMenuRef(ref) {
    this.menu = ref;
  }

  showMenu() {
    this.menu.show();
  }

  selectOption(option) {
    const { handlePress } = this.props;

    if (handlePress) {
      handlePress(option);
    }
    this.hideMenu();
  }

  hideMenu() {
    this.menu.hide();
  }

  render() {
    const {
      iconName,
      iconStyle,
      iconContainerStyle,
      containerStyle,
      options,
      itemTextStyle,
      itemContainerStyle,
      buttonTestID,
    } = this.props;

    return (
      <Menu
        ref={this.setMenuRef}
        button={
          <MenuButton
            handlePress={this.showMenu}
            iconName={iconName}
            iconStyle={iconStyle}
            style={iconContainerStyle}
            testID={buttonTestID}
          />
        }
        style={[styles.container, containerStyle]}
      >
        {options &&
          options.map((option) => {
            return (
              <MenuItem
                testID={`menu.item.${option.text}`}
                key={option.text}
                onPress={() => this.selectOption(option)}
                disabled={option.disabled}
                textStyle={[styles.itemText, itemTextStyle]}
                style={[styles.itemContaier, itemContainerStyle]}
              >
                {option.text}
              </MenuItem>
            );
          })}
      </Menu>
    );
  }
}
