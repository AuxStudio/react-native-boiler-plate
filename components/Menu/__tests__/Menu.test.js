import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Menu from '..';

const OPTIONS = [
  {
    text: 'Test',
  },
  {
    text: 'Disabled',
    disabled: true,
  },
];

describe('Menu', () => {
  let spy;

  it('renders with all props', () => {
    expect(
      renderer.create(
        <Menu
          options={OPTIONS}
          handlePress={jest.fn()}
          iconName="chevron-left"
          iconStyle={{ color: 'red' }}
          iconContainerStyle={{ backgroundColor: 'blue' }}
          itemTextStyle={{ color: 'red' }}
          itemContainerStyle={{ backgroundColor: 'blue' }}
          containerStyle={{ backgroundColor: 'green' }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders a Menu with minimum required props', () => {
    expect(renderer.create(<Menu options={OPTIONS} handlePress={jest.fn()} />)).toMatchSnapshot();
  });

  it('calls setMenuRef on componentDidMount', () => {
    spy = jest.spyOn(Menu.prototype, 'setMenuRef');

    renderer.create(
      <Menu
        options={OPTIONS}
        handlePress={jest.fn()}
        itemTextStyle={{ color: 'red' }}
        itemContainerStyle={{ backgroundColor: 'blue' }}
        containerStyle={{ backgroundColor: 'green' }}
      />,
    );
    expect(spy).toHaveBeenCalled();
  });

  afterEach(() => {
    if (spy) {
      spy.mockClear();
    }
  });
});
