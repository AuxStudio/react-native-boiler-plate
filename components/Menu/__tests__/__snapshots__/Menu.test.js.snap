// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders a Menu with minimum required props 1`] = `
<View
  collapsable={false}
>
  <View
    onLayout={[Function]}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": undefined,
                "fontSize": 24,
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 2,
            "left": 0,
            "opacity": 0,
            "paddingVertical": 8,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.14,
            "shadowRadius": 2,
            "top": 0,
            "transform": Array [],
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "maxWidth": 248,
                  "minWidth": 124,
                },
                Array [
                  undefined,
                  undefined,
                ],
                Object {
                  "width": 0,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 14,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                  },
                  false,
                  Array [
                    Object {},
                    undefined,
                  ],
                ]
              }
            >
              Test
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "maxWidth": 248,
                  "minWidth": 124,
                },
                Array [
                  undefined,
                  undefined,
                ],
                Object {
                  "width": 0,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 14,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                  },
                  Object {
                    "color": "#BDBDBD",
                  },
                  Array [
                    Object {},
                    undefined,
                  ],
                ]
              }
            >
              Disabled
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Menu renders with all props 1`] = `
<View
  collapsable={false}
>
  <View
    onLayout={[Function]}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "blue",
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": undefined,
                "fontSize": 24,
              },
              Object {
                "color": "red",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "green",
            "borderRadius": 2,
            "left": 0,
            "opacity": 0,
            "paddingVertical": 8,
            "position": "absolute",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.14,
            "shadowRadius": 2,
            "top": 0,
            "transform": Array [],
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "maxWidth": 248,
                  "minWidth": 124,
                },
                Array [
                  undefined,
                  Object {
                    "backgroundColor": "blue",
                  },
                ],
                Object {
                  "width": 0,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 14,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                  },
                  false,
                  Array [
                    Object {},
                    Object {
                      "color": "red",
                    },
                  ],
                ]
              }
            >
              Test
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "maxWidth": 248,
                  "minWidth": 124,
                },
                Array [
                  undefined,
                  Object {
                    "backgroundColor": "blue",
                  },
                ],
                Object {
                  "width": 0,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 14,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                  },
                  Object {
                    "color": "#BDBDBD",
                  },
                  Array [
                    Object {},
                    Object {
                      "color": "red",
                    },
                  ],
                ]
              }
            >
              Disabled
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
